content/accessible/labels-and-text-alternatives/index.md

```diff
@@ -0,0 +1,334 @@
+    …
+  &lt;/body&gt;
+&lt;/html&gt;
+</pre>
+
+<div class="aside note">
+For tips on writing effective titles, see the
+<a href="https://web.dev/discoverable/write-descriptive-text">
+Write descriptive titles guide
+</a>.
+</div>
+
+Similarly, any `frame` or `iframe` elements should have a `title` attribute.
+
+```html
+<iframe title="An interactive map of San Francisco" src="…"></iframe>
+```
+
+While an `iframe` may contain its own internal `title` element, a screen
+reader will usually stop at the frame boundary and announce the element's
+role—"frame"—and its accessible name, provided by the `title` attribute. This
+lets the user to decide if they wish to enter the frame or bypass it.
+
+## Text alternatives for images and objects
+An `img` should always be accompanied by an
+[`alt`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#Attributes)
+attribute, to give the image its accessible name. If the image fails to load,
+the `alt` text will be used as a placeholder so users have a sense of what the
+image was trying to convey.
+
+Writing good `alt` text is a bit of an art, but there are a couple of guidelines
+you can follow:
+
+1. Determine if the image provides content that would otherwise be difficult to
+attain from reading the surrounding text.
+1. If so, convey the content as succinctly as possible.
+
+If the image is more of a decoration, and does not provide any useful content,
+you can give it an empty `alt=""` attribute to remove it from the accessibility
+tree.
+
+<div class="aside note">
+You can learn more about writing effective <code>alt</code> by checking out <a
+href="https://webaim.org/techniques/alttext/">WebAIM's guide to Alternative
+Text.</a>
+</div>
+
+### Images as links and inputs
+
+An image wrapped in a link should describe where the user will navigate to if
+they click the link.
+
+```html
+<a href="https://en.wikipedia.org/wiki/Google">
+  <img alt="Google's wikipedia page" src="google-logo.jpg">
+</a>
+```
+
+Similarly, if an `<input type="image">` element is being used to create an image
+button, it should contain `alt` text that describes the action that will occur
+if the user clicks the button.
+
+<pre class="prettyprint devsite-disable-click-to-copy">
+&lt;form&gt;
+  &lt;label&gt;
+    Username:
+    &lt;input type=&quot;text&quot;&gt;
+  &lt;/label&gt;
+  <strong>&lt;input type=&quot;image&quot; alt=&quot;Sign in&quot;</strong>
+             <strong>src=&quot;./sign-in-button.png&quot;&gt;</strong>
+&lt;/form&gt;
+</pre>
+
+### Embedded objects
+
+`<object>` elements, which are typically used for embeds like Flash, PDFs, or
+ActiveX, should also contain alternative text. Similar to images, this text will
+display if the element fails to render.
+
+```
+<object type="application/pdf"
+    data="/report.pdf"
+    width="600"
+    height="400">
+Annual report.
+</object>
+```
+
+## Label buttons and links
+Buttons and links are simple, and common, interactive elements. Because they are
```

Comment:
Suggestion to reword the second sentence a little bit, as I tend to avoid starting sentences with 'because'. Maybe combine crucial with first sentence. Something like:

"Buttons and links are simple, and common, interactive elements; they are also crucial to the experience of a site. It's important that both have good accessible names."